### PR TITLE
Mapping/dynamic mortar coupling

### PR DESCRIPTION
**Pull request by peterjwilson**
**Description**
Progress to partition couple fem-fem/mpm with feti-type approach.

### CODE REVIEW

File: applications/CoSimulationApplication/custom_utilities/feti_dynamic_coupling_utilities.cpp [hunk 1]
```diff
@@ -16,74 +16,294 @@
...
 {
+    FetiDynamicCouplingUtilities::FetiDynamicCouplingUtilities(ModelPart& rInterfaceOrigin,
+        ModelPart& rInterFaceDestination,
+        double OriginNewmarkBeta, double OriginNewmarkGamma,
+        double DestinationNewmarkBeta, double DestinationNewmarkGamma)
+    :mrOriginInterfaceModelPart(rInterfaceOrigin), mrDestinationInterfaceModelPart(rInterFaceDestination),
+        mOriginBeta(OriginNewmarkBeta), mOriginGamma(OriginNewmarkGamma),
+        mDestinationBeta(DestinationNewmarkBeta), mDestinationGamma(DestinationNewmarkGamma)
+    {
+        KRATOS_WATCH("11111");
```
Comment (loumalouomega) [inline applications/CoSimulationApplication/custom_utilities/feti_dynamic_coupling_utilities.cpp]:
```suggestion
```
Leftover

File: applications/MappingApplication/custom_mappers/mapper.h [hunk 2]
```diff
@@ -160,7 +160,10 @@ class Mapper
      * @return The mapping-matrix
      */
     virtual TMappingMatrixType* pGetMappingMatrix() = 0;
-    virtual DenseMappingMatrixUniquePointerType pGetDenseMappingMatrix();
+    virtual DenseMappingMatrixUniquePointerType pGetDenseMappingMatrix()
+    {
+        KRATOS_ERROR << "YEEEEEEEEEEEEEEEEEE";
```
Comment (loumalouomega) [inline applications/MappingApplication/custom_mappers/mapper.h]: Better comment?¿

**Review by loumalouomega — COMMENTED** [review]
Just curious, how this mortar implementation works?

**Comment by loumalouomega** [comment]
Still curious how this mortar works

**Comment by peterjwilson** [comment]
> 
> 
> Still, curious how this mortar works

Hey Vicente! Glad you're interested 😄 . I've seen your very impressive contact mortar work, which I am guessing is done with a monolithic solve?

This branch is a continuation of the recent mortar implemented in the mappers app, intended for a partitioned CoSim solve. The goal is to couple two structural parts together (which, in the future may hopefully be different numerical methods) for dynamic analysis. Partitioned gauss seidel coupling of similar stiffnesses gives problems (added mass effect in fsi), so this branch takes a feti-based method (enforcing interface velocity continuity via lagrange multipliers) and extends it to non-conforming meshes with mortar mapping.

It's still just a work in progress - any thoughts you have would be cool to hear.

**Pull request closed without merging**